runtime/lib/ttnn/program.cpp

```diff
@@ -195,65 +196,70 @@ run(::tt::target::ttnn::EmptyOp const *op, ::ttnn::Device &device,
   liveTensors.try_emplace(op->out()->global_id(), &tensorPool.back());
 }
 
+static void
+runEltwiseBinaryOP(
```

Comment:
@mrakitaTT thanks! This definitely removes alot of boilerplate code, but my worry is that we may want different configs/parameters for different ops. For example I see that binary ops in ttnn accept scalar values; there's also implicit broadcasting for associative binary ops, which may or may not affect the constraints on the memory configs of the input tensors. Therefore I'm leaning towards leaving the structure unchanged for now, and once we have well-documented op constraints then we can merge/generalize the code. What do you think?

Reply:
I like this refactor. @jnie-TT, we can easily pass additional configs through these new functions if needed, right?

Reply:
@jnie-TT Thanks for pointing that out, I understand the concern, but IMO it is always better to start with generalized compact code and then only if it turns out that we need bunch of specific customizations to go from there and expand the code. As @sdjordjevicTT said it might be fairly easy to pass additional configs to these functions, but even if it ends up being not so easy for certain OPs then we can have custom functions for those OPs only. Budabackend stack was littered with duplicated code and I would really like us to change that mindset for the future stacks :)

Reply:
Hi @jnie-TT do you still have concerns about this? I would like to proceed with merge, just waiting on your sign-off.